src/transformers/models/mamba2/convert_mamba2_ssm_checkpoint_to_pytorch.py

```diff
@@ -15,55 +15,130 @@
+    config_path = (
+        path.join(config_path, "params.json")
+        if path.isfile(path.join(config_path, "params.json"))
+        else path.join(config_path, "config.json")
+    )
+    with open(config_path, "r", encoding="utf-8") as json_file:
+        config = json.load(json_file)
+    hf_config, is_not_codestral = convert_ssm_config_to_hf_config(config)
+    hf_config.save_pretrained(output_dir)
+
+    # Load state dict of the original model
+    state_dict_load_function = load_state_dict_from_torch if is_not_codestral else load_state_dict_from_safetensors
+    original_state_dict = state_dict_load_function(mamba2_checkpoint_path)
+
+    # Load and transfer to hf model
+    hf_model = Mamba2ForCausalLM(hf_config)
     hf_model.load_state_dict(original_state_dict)
 
     # Save new model to pytorch_dump_path
-    hf_model.to(torch.bfloat16).save_pretrained(output_dir)
-    tokenizer_class = LlamaTokenizerFast
-    tokenizer = tokenizer_class(tokenizer_model_path, legacy=False, from_slow=True)
+    dtype = torch.float32 if precision == "fp32" else (torch.bfloat16 if precision == "bf16" else torch.float16)
+    save_model(hf_model.to(dtype), path.join(output_dir, "model.safetensors"), metadata={"format": "pt"})
+
+    # Load and save tokenizer
+    if tokenizer_model_path is not None and not is_not_codestral:
+        tokenizer_class = LlamaTokenizerFast
+        tokenizer = tokenizer_class(tokenizer_model_path, legacy=False, from_slow=True)
+    else:
+        tokenizer = AutoTokenizer.from_pretrained("state-spaces/mamba-130m-hf")
+        # TODO: how to correctly overwrite padding side so it is saved as is
+        tokenizer.padding_side = "left"
     tokenizer.save_pretrained(output_dir)
 
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
     parser.add_argument(
         "-i",
-        "--mamba2_checkpoint_file",
+        "--mamba2_checkpoint_directory",
         type=str,
         required=True,
-        help="Path to a `pytorch_model.bin` mamba2_ssm checkpoint file to be converted.",
+        help="Path to a directory containing the `pytorch_model.bin` or `.safetensors` mamba2_ssm checkpoint file to be converted.",
     )
     parser.add_argument(
-        "-c",
-        "--tokenizer_model_path",
+        "-p",
+        "--precision",
         type=str,
+        default="bf16",
         required=True,
-        help="Path to a `config.json` file corresponding to a Mamba2Config of the original mamba2_ssm model.",
+        choices=["fp32", "fp16", "bf16"],
+        help="The precision the model will be saved in. Select from fp32, fp16 or bf16.",
     )
     parser.add_argument(
         "-o", "--output_dir", type=str, required=True, help="Path to directory to save the converted output model to."
     )
+    parser.add_argument(
+        "-t",
+        "--tokenizer_model_path",
+        type=str,
+        default=None,
+        required=False,
+        help="Path to a tokenizer file.",
+    )
```

Comment:
Was it pointing to https://huggingface.co/mistralai/Mamba-Codestral-7B-v0.1/blob/main/tokenizer.model.v3 ? 

Might as well automate it based on using the directory.

Reply:
> Was it pointing to https://huggingface.co/mistralai/Mamba-Codestral-7B-v0.1/blob/main/tokenizer.model.v3 ?

Yes, this script in master branch works out-of-box for Mamba-Codestral

Reply:
Sounds good to me. Let's leave it there :+1: